pkg/adapters/http.go

```diff
@@ -0,0 +1,81 @@
+package adapters
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/weaveworks/weave-gitops/pkg/templates"
+)
+
+const (
+	expiredHeaderName = "Entitlement-Expired-Message"
+)
+
+// An HTTP client of the cluster service.
+type HttpClient struct {
+	baseURI *url.URL
+	client  *resty.Client
+}
+
+// NewHttpClient creates a new HTTP client of the cluster service.
+// The endpoint is expected to be an absolute HTTP URI.
+func NewHttpClient(endpoint string, client *resty.Client, out io.Writer) (*HttpClient, error) {
```

Comment:
This `adapters` package is only relevant for `templates`, correct? If so, should we put this code in `templates`?

Reply:
Currently yes but it would soon be used to retrieve `clusters` too. The intent of that package is to keep any adapter specific code (HTTP/gRPC etc.) in one place separate from the command/presentation logic. The two packages `adapters` and `templates` are then used together when the command is invoked. Later on, another package could be added i.e. `clusters` that deals with CAPI cluster commands and that could also be used together with the `adapters` package to retrieve and display cluster information.

I don't feel too strongly about this though, let me know if you think it complicates things.